ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/block/AbstractBlockProcessor.java

```diff
@@ -370,6 +398,34 @@ public void processAttestations(
     verifyAttestations(state, attestations, BLSSignatureVerifier.SIMPLE, indexedAttestationCache);
   }
 
+  @Override
+  public void processAttestationsNoValidation(
+      final MutableBeaconState state, final SszList<Attestation> attestations)
+      throws BlockProcessingException {
+    try {
+      for (Attestation attestation : attestations) {
+        // Validate
+        assertAttestationValid(state, attestation);
```

Comment:
I'm slightly confused that a method with `NoValidation` in the name is performing validation.

Reply:
Yes, all of the "no validation" naming across these methods is very confusing.  I at least added some javadoc to explain what the "no validation" means.  But would probably be clearer if we just passed a boolean that toggles the targeted validation on / off.  Will look into that in a follow-up.